src/syngen/ml/vae/wrappers/wrappers.py

```diff
@@ -370,9 +370,15 @@ def _train(self, dataset, epochs: int):
                 break
             epoch += 1
 
-    def _create_optimizer(self):
-        learning_rate = 1e-04 * np.sqrt(self.batch_size / BATCH_SIZE_DEFAULT)
-        return tf.keras.optimizers.Adam(learning_rate=learning_rate)
+    def _create_optimizer(self, learning_rate=None):
+        if not learning_rate:
+            learning_rate = 1e-04 * np.sqrt(self.batch_size / BATCH_SIZE_DEFAULT)
+        import platform
```

Comment:
I think that it's better to encapsulate the next logic in the separate method '_create_optimizer' - 
`
@staticmethod
def _create_optimizer(learning_rate):
import platform
        if platform.processor() == 'arm':
            logger.info('Mac ARM processor is detected. Legacy Adam optimizer has been created.')
            return tf.keras.optimizers.legacy.Adam(learning_rate=learning_rate)
        else:
            return tf.keras.optimizers.Adam(learning_rate=learning_rate)` Then rename the current method '_create_optimizer' to '__create_optimizer', and call '_create_optimizer' inside '__create_optimizer'